packages/worker/src/worker.ts

```diff
@@ -618,10 +618,11 @@ export class Worker {
                     // NOTE: activity will not be considered cancelled until it confirms cancellation
                     this.log.debug('Cancelling activity', { taskToken: base64TaskToken });
                     span.setAttribute('found', true);
-                    if (!task.cancel?.reason) {
+                    const reason = task.cancel?.reason;
+                    if (reason === undefined || reason === null) {
```

Comment:
Is this just because it was an empty string?

Reply:
The enum value for `NOT_FOUND` is `0`, I overlooked that.